<launch>

    <arg name="cmd_src"         default="circle"/>
    <arg name="robot"           default="nusim" />
    <arg name="use_rviz"        default="true"  />

    <param name="odom_id" type="string" value="odom"/>
    <param name="body_id" type="string" value="blue_base_footprint"/>


    <arg name="color" value="red"/>

    <node name="nusim" pkg="nusim" type="nusim_node" />

    <rosparam file="$(find nusim)/config/basic_world.yaml" />

    <include file="$(find nuturtle_description)/launch/load.launch" pass_all_args="true">
        <arg name="color"       value="red"/>
        <arg name="use_jsp"     value="true"/>
        <arg name="use_rviz"    value="false"/>
    </include>

    <node pkg="rviz" type="rviz" name="rviz" required="true" args="-f world -d $(find nusim)/config/nusim.rviz" ns="$(arg color)"/>

    <node pkg="tf2_ros" type="static_transform_publisher" name="link1_broadcaster" args="0 0 0 0 0 0 0 world odom" />

    <node name="odometry" pkg="nuturtle_control" type="odometry"/>
    <node name="turtle_interface" pkg="nuturtle_control" type="turtle_interface"/>

    <node name="circle" pkg="nuturtle_control" type="circle" if="$(eval arg('cmd_src')=='circle')"/>


</launch>